### DISCUSSION TITLE
InfluxDB driver add string support?

### DISCUSSION THREAD

**Discussion by pkkrusty** [arendst/Tasmota#18954] [Support request] [answered]
Currently, influxdb driver checks for numeric data (plus ON/OFF/TOGGLE) before it puts things into the database. However, influxDB does accept string values, which can be useful for things like Device Status or Error displayed along with numeric graphed data in Grafana.  How easy would it be to add string values to influxdb driver?

The possibility exists to do value mappings in Grafana, but not as clean.

**Comment by barbudor** [comment]
I wonder why the author as decided to publish only numbers...

Not sure to fully understand your use-case.
Are you looking to support textual value in addition of what currently exist (which means leave the conversion of StateText to StateValue but support textual values for other type of values (like "door":"open")
Or to completely avoid the conversion of OFF=>0, ON=>1, ...

(personally I'm not using this driver but do all mapping of SENSOR to InfluxDb in NodeRed)

**Reply by sfromis** [reply]
I suspect that a reasoning for focusing on numeric values is that they are the ones to produce stats/graphs on, most text values are rather "boring" (static), like Wifi info or textual uptime (in addition to seconds), and does add quite a bit to the amount of data forwarded. If adding text, an option similar to `IfxSensor` could make sense for specialized use cases with text.

**Reply by pkkrusty** [reply]
My specific use-case is Status and Errors of some inverters, which I'm pulling from Modbus as numbers, then doing the mapping to text in scripting then publishing. Of course all that can easily be done in Node-Red or Grafana. True that it is less traffic and overhead for the ESP8266 to deal with to just publish the number codes. 
![image](https://github.com/arendst/Tasmota/assets/79770016/db7f0e79-5d9b-4e85-ae7e-455636f5b0be)

**Comment by arendst** [comment, marked as answer]
The current solution tests for numeric (or OFF, ON, TOGGLE etc.) to lower comms to influxdb. Otherwise every JSON data pair would flood the database (and comms with lots of non-information).

**Reply by pkkrusty** [reply]
Seems reasonable. I can do the back-end processing.
